go/sys/remote/http_archive.go

```
@@ -0,0 +1,458 @@
+	if strings.ContainsRune(entry, 0) {
+		return "", fmt.Errorf("%w: entry %q contains NUL", ErrUnsafeDestination, entry)
+	}
+	// Component-level traversal check on the raw, pre-normalisation
+	// form: any ".." segment means the archive intended to escape.
+	// Split on either separator so a Windows-style entry in a tar
+	// can't slip past on a Unix host.
+	for _, comp := range strings.FieldsFunc(entry, func(r rune) bool {
+		return r == '/' || r == '\\'
+	}) {
+		if comp == ".." {
+			return "", fmt.Errorf("%w: entry %q contains ..", ErrUnsafeDestination, entry)
```

golangci-lint (2.12.2)

[error] ST1005: error strings should not end with punctuation or newlines

(staticcheck)